artemis/modules/sql_injection_detector.py

```diff
@@ -0,0 +1,187 @@
+
+        return flag
+
+    @staticmethod
+    def contains_error(response: HTTPResponse) -> bool | None:
+        for message in SQL_ERROR_MESSAGES:
+            if re.search(message, response.content):
+                return True
+        return False
+
+    @staticmethod
+    def response_when_injecting(url: str) -> HTTPResponse:
+        response = http_requests.get(url)
+
+        return response
+
+    def scan(self, urls: List[str], task: Task) -> Dict[str, object]:
+        task_result = {
+            "task_host": get_target_url(task),
+            "status": task.status,
+            "message": [],
+        }
+        message = []
+        for current_url in urls:
+            #  Below is the code describing the logic, but it is repetitive. What is better (create a universal
+            #  function or leave it as below)
+            if self.is_url_with_payload(current_url):
+                url_with_payload = self.change_url_params(url=current_url)
+
+                if not self.contains_error(self.response_when_injecting(current_url)) and self.contains_error(
+                    self.response_when_injecting(url_with_payload)
+                ):
+                    message.append(f"{url_with_payload}: It appears that this url is vulnerable to SQL Injection")
+
+                for sleep_payload in Config.Modules.SqlInjectionDetector.SLEEP_PAYLOAD:
+                    url_with_sleep_payload = self.change_url_params(url=current_url, sleep_payload=sleep_payload)
+
+                    if self.are_requests_time_efficient(current_url) and not self.are_requests_time_efficient(
+                        url_with_sleep_payload
+                    ):
+                        message.append(
+                            f"{url_with_sleep_payload}: It appears that this url is vulnerable to SQL Time Base Injection"
+                        )
+            else:
+                for param_batch in more_itertools.batched(URL_PARAMS, 10):
+                    url_with_payload = self.create_url_with_batch_payload(url=current_url, param_batch=param_batch)
+
+                    if not self.contains_error(self.response_when_injecting(current_url)) and self.contains_error(
+                        self.response_when_injecting(url_with_payload)
+                    ):
+                        message.append(f"{url_with_payload}: It appears that this url is vulnerable to SQL Injection")
+
+                    for sleep_payload in Config.Modules.SqlInjectionDetector.SLEEP_PAYLOAD:
+                        url_with_sleep_payload = self.create_url_with_batch_payload(
+                            url=current_url, param_batch=param_batch, sleep_payload=sleep_payload
+                        )
+
+                        if self.are_requests_time_efficient(current_url) and not self.are_requests_time_efficient(
+                            url_with_sleep_payload
+                        ):
+                            message.append(
```

Comment:
repeat that check (e.g. 3 times) to make sure it's not a FP

Reply:
Done.

Reply:
Done.

Reply:
btw, don't cal this varialbe message, it's not a string messae